Set more padding on volume pref SeekBar for Lollipop and above

johnheroy opened the pull request:
Improves the visual appearance of the VolumePreference Seekbar for Lollipop and above. See pics for below for screenshots across APIs 10, 15, and 23 post this change. Fixes #210.

# API v23
![api v23](https://cloud.githubusercontent.com/assets/5909572/10616742/8fb7e2ac-771b-11e5-8748-eda7de465829.png)

# API v15
![api v15](https://cloud.githubusercontent.com/assets/5909572/10616741/8def86a0-771b-11e5-9558-a3dc31ea0171.png)

# API v10
![api v10](https://cloud.githubusercontent.com/assets/5909572/10616739/8be53e18-771b-11e5-901f-fd913357b279.png)

jklein24 commented on app/src/main/java/org/connectbot/util/VolumePreference.java:
```diff
@@ -56,7 +57,13 @@ protected View onCreateDialogView() {
-		sb.setPadding(10, 10, 10, 10);
+
+		if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+			sb.setPadding(75, 70, 75, 10);
```
Did this come from the material spec somewhere?

johnheroy replied:
No this did not come from the material spec, was just adjusting the padding so that the Seekbar has even padding and not hugging the top edge of the dialog.

Looking again at the [material spec](https://www.google.com/design/spec/components/sliders.html#sliders-continuous-slider) though we could do better by adding a volume icon to the left of the Seekbar. I could add this in another commit and add to the PR if you would like that change :)

kruton replied:
Could this be moved to a resource? I didn't realize this was in the java code. It'd be best to put it in an XML layout, but it looks like this doesn't lend itself to that. Instead you could just add a `<dimen>` resource.

In a resource file such as `res/values/dimens.xml` you could add:
```
<dimen name="volume_dialog_padding_horizontal">75dp</dimen>
<dimen name="volume_dialog_padding_vertical">10dp</dimen>
```

Then you could do:
```
int paddingHorizontal = (int) getResources().getDimension(R.dimen.volume_dialog_padding_horizontal);
int paddingVertical = (int) getResources().getDimension(R.dimen.volume_dialog_padding_vertical);
sb.setPadding(paddingHorizontal, paddingVertical, paddingHorizontal, paddingVertical);
```

jklein24 replied:
Better yet:

getContext().getResources().getDimension(android.support.design.R.dimen.abc_select_dialog_padding_start_material)

and

getContext().getResources().getDimension(android.support.design.R.dimen.abc_dialog_list_padding_vertical_material)

jklein24 commented:
The padding specs for a dialog are here:

https://www.google.com/design/spec/components/dialogs.html#dialogs-specs

It looks like 75px is roughly 24dp in your screenshot, but that might not be true on all devices. I'd make sure it's 24dp.

johnheroy commented:
I added the volume icon to match the slider examples in the material docs and matched the appropriate spacing (24dp to left of icon, content 80dp from left of dialog). To make this easier I factored out the dialog into an XML layout file, screenshots for the various versions as below:

# API v23
![api v23](https://cloud.githubusercontent.com/assets/5909572/10623548/c6453c2e-7744-11e5-95ea-954ce8fff379.png)

# API v15
![api v15](https://cloud.githubusercontent.com/assets/5909572/10623557/cb8a9a44-7744-11e5-9811-dd53066d723d.png)

# API v10
![api v10](https://cloud.githubusercontent.com/assets/5909572/10623560/d08fe012-7744-11e5-9b41-b6dd9858ffb2.png)

johnheroy commented:
I'm not totally sure why the [Travis CI build](https://travis-ci.org/connectbot/connectbot/builds/86517038) is failing, I can just see that there was a lint error somewhere at my ImageView declaration but not sure what...is there a way I can run the linter locally and view the precise error?

In any case I have a hunch it could be because I'm missing the "android:contentDescription" attribute. Is the general practice to add a string first (something like "Bell volume") to the English strings.xml and then it would be translated later?

jklein24 commented:
Unfortunately, there's not a great process with the linter. You'll nee to run `./gradlew app:lint` and then check out the output html. You should be able to track down the lint errors in the files you've changed here.

On another note, sorry to nit-pick but the space between the icon and slider seems too big to me for some reason. Is there extra padding there on accident or something?

kruton commented:
Yes, if you run `./gradlew lint` then look at `app/bplid/outputs/lint-results.html` you'll see the error is:
```
[Accessibility] Missing contentDescription attribute on image
```

alescdb commented:
Just a trick (hint?) : You may also start lint from Android Studio with menu Analyze -> Inspect Code, travis-ci build gives you the file and the line where the new lint warning/error is.
In your case : 

```
$ ./scripts/check-lint-count.sh app/build/outputs/lint-results.xml $HOME/.cache/lint/lint-results.xml
Historical count : 552, new count : 553
FAILURE: lint issues increased from 552 to 553
497a498
> <location file="/home/travis/build/connectbot/connectbot/app/src/main/res/layout/volume_preference_dialog_layout.xml" line="29" column="5"/>
```

johnheroy commented:
Thanks for the catch, it looked like there was some inner padding on the SeekBar in Lollipop itself so I just reset with android:padding="0dp" for that element and added a contentDescription for the ImageView for a11y. Updated screencaps below:

# API v23
![api v23](https://cloud.githubusercontent.com/assets/5909572/10643054/b5916e66-77d5-11e5-86b7-d778a61c7fdd.png)

# API v14
![api v14](https://cloud.githubusercontent.com/assets/5909572/10643063/bdd10e6a-77d5-11e5-9070-13227d16ecf9.png)

# API v10
![api v10](https://cloud.githubusercontent.com/assets/5909572/10643066/c3be838e-77d5-11e5-8aad-742c20d8b06f.png)

jklein24 commented on app/src/main/java/org/connectbot/util/VolumePreference.java:
```diff
@@ -46,20 +48,17 @@ public VolumePreference(Context context, AttributeSet attrs, int defStyle) {
-
-		return sb;
+		volumeBar.setOnSeekBarChangeListener(this);
+		super.onBindDialogView(view);
```
Should this be done first in the method?

johnheroy replied:
I had seen a lot of examples calling the super class's method at the end but tested both ways and does not seem to have made any difference, for convention I moved to the first line of the method.

kruton commented:
I just updated the lint failure output so it's more clear what is causing the issue.

johnheroy commented:
FYI, commit 9bceeb in conjunction with these changes made the seekbar styling on API v10 look consistent with v14/15 and v23. See below:

# API v23
![api v23](https://cloud.githubusercontent.com/assets/5909572/10651329/c6053236-7803-11e5-869c-25b57b8ab61a.png)

# API v15
![api v15](https://cloud.githubusercontent.com/assets/5909572/10651336/cd4ff9cc-7803-11e5-960c-20eae35f67f8.png)

# API v10
![api v10](https://cloud.githubusercontent.com/assets/5909572/10651356/e7153a3e-7803-11e5-892c-6f8607a620ed.png)
